mongorestore/restore.go

```diff
@@ -332,26 +332,87 @@ func (restore *MongoRestore) RestoreIntent(intent *intents.Intent) Result {
 	return result
 }
 
-func convertLegacyIndexes(indexes []IndexDocument) []IndexDocument {
-	for i, index := range indexes {
-		for j, elem := range index.Key {
-			switch elem.Value {
-			case true:
-				indexes[i].Key[j].Value = 1
-				log.Logvf(log.Always, "convertLegacyIndexes: converting the legacy vaule of '%s' from '%s' to '%s' in index '%s' on collection '%s'",
-					indexes[i].Key[j].Key, "true", "1", indexes[i].Options["name"], indexes[i].Options["ns"])
+func convertLegacyIndexes(indexes []IndexDocument) {
+	for _, index := range indexes {
+		convertLegacyIndexKeys(index)
+		convertLegacyIndexOptions(index)
+	}
+}
+
+func convertLegacyIndexKeys(index IndexDocument) {
+	var converted bool
+	originalJSONString := createExtJSONString(index.Key)
```

Comment:
you can move this into the "if converted" stanza, as well, as it's only needed conditionally.

Reply:
But index.Key is a slice and I'm editing the underlying array elements. I think this is less intrusive than making a full copy of the array.

Reply:
right. nevermind. :-)